android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.java

```diff
@@ -493,6 +510,46 @@ public void handleIntent(Activity activity, Intent intent) {
       }
     }
 
+    openDefaultUrl();
+  }
+
+  private boolean openExperienceFromNotificationIntent(Intent intent) {
+    NotificationResponse response = unmarshallNotificationResponse(intent.getByteArrayExtra(NOTIFICATION_RESPONSE_KEY));
+    String experienceIdString = ScopedNotificationsUtils.getExperienceId(response);
+    if (experienceIdString == null) {
+      return false;
+    }
+
+    ExperienceDBObject experience = ExponentDB.experienceIdToExperienceSync(experienceIdString);
+    if (experience == null) {
+      Log.w("expo-notifications", "Couldn't find experience from experienceId.");
+      return false;
+    }
+
+    String manifestUrl = experience.manifestUrl;
+    openExperience(new KernelConstants.ExperienceOptions(manifestUrl, manifestUrl, null));
+    return true;
+  }
+
+  @Nullable
+  private NotificationResponse unmarshallNotificationResponse(@Nullable byte[] notificationRequestByteArray) {
+    if (notificationRequestByteArray == null) {
+      return null;
+    }
+    try {
+      Parcel parcel = Parcel.obtain();
+      parcel.unmarshall(notificationRequestByteArray, 0, notificationRequestByteArray.length);
+      parcel.setDataPosition(0);
+      NotificationResponse response = NotificationResponse.CREATOR.createFromParcel(parcel);
+      parcel.recycle();
+      return response;
+    } catch (Exception e) {
+      Log.e("expo-notifications", "Could not unmarshall NotificationResponse from Intent.extra.", e);
+    }
+    return null;
+  }
```

Comment:
Let's move this function to `NotificationResponseReceiver` and make it `public` — if we consider putting the notification response inside the Intent a sort of public developer API (as we do since we also allow assigning `expo.modules.notifications.OPEN_APP_ACTION` to a custom developer-picked Activity) we should also expose a nice API for retrieving this response. This could even let us not import `NOTIFICATION_RESPONSE_KEY` from `NotificationResponseReceiver`, but just call something like `getNotificationResponse(intent)` which would return `null` if there was no response. 🤔